elasticsearch_dsl/aggs.py

```diff
@@ -70,48 +90,58 @@ def A(name_or_agg, filter=None, **params):
 class Agg(DslBase):
     _type_name = "agg"
     _type_shortcut = staticmethod(A)
-    name = None
+    name = ""
 
-    def __contains__(self, key):
+    def __contains__(self, key: str) -> bool:
         return False
 
-    def to_dict(self):
+    def to_dict(self) -> Dict[str, JSONType]:
         d = super().to_dict()
-        if "meta" in d[self.name]:
-            d["meta"] = d[self.name].pop("meta")
+        if isinstance(d[self.name], dict):
+            n = cast(Dict[str, JSONType], d[self.name])
+            if "meta" in n:
+                d["meta"] = n.pop("meta")
         return d
 
-    def result(self, search, data):
+    def result(self, search: "SearchBase", data: Any) -> AttrDict[str, Any]:
         return AggResponse(self, search, data)
 
 
 class AggBase:
-    _param_defs = {
+    if TYPE_CHECKING:
+        aggs: Dict[str, Agg]
+        _base: Agg
+    _params: Dict[str, Any]
```

Comment:
Why isn't `params` under `if TYPE_CHECKING:`? (By the way, I don't usually bother with such guards unless it avoids an import at runtime or fixes a cyclic dependency)

Reply:
You are right, this is inconsistent and in this case there is no point in having a guard.